tests/integration/cmd_push_test.go

```diff
@@ -0,0 +1,209 @@
+package integration
+
+import (
+	"fmt"
+	"github.com/openshift/odo/tests/helper"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("odo push command tests", func() {
+	var project string
+	var context string
+
+	appName := "app"
+	cmpName := "nodejs"
+
+	oc = helper.NewOcRunner("oc")
+
+	// This is run after every Spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		project = helper.CreateRandProject()
+		context = helper.CreateNewContext()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.DeleteProject(project)
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("when push command is executed", func() {
+		It("should not build when no changes are detected in the directory and build when a file change is detected", func() {
+			helper.CmdShouldPass("git", "clone", "https://github.com/openshift/nodejs-ex", context+"/nodejs-ex")
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context+"/nodejs-ex", "--app", appName)
+			helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+			helper.CmdShouldPass("odo", "url", "create", "--port", "8080", "--context", context+"/nodejs-ex")
+			output := helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+
+			Expect(output).To(ContainSubstring("No file changes detected, skipping build"))
+
+			url := oc.GetFirstURL(cmpName, appName, project)
+			helper.ReplaceString(filepath.Join(context+"/nodejs-ex"+"/views/index.html"), "Welcome to your Node.js application on OpenShift", "UPDATED!")
+
+			helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+			helper.HttpWaitFor("http://"+url, "UPDATED!", 30, 1)
+		})
+
+		It("should build when a new file and a new folder is added in the directory", func() {
+			helper.CmdShouldPass("git", "clone", "https://github.com/openshift/nodejs-ex", context+"/nodejs-ex")
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context+"/nodejs-ex", "--app", appName)
+			helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+			output := helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+
+			Expect(output).To(ContainSubstring("No file changes detected, skipping build"))
+
+			newFilePath := filepath.Join(context, "nodejs-ex", "new-example.html")
+			if err := helper.CreateFileWithContent(newFilePath, "<html>Hello</html>"); err != nil {
+				fmt.Printf("the .odoignore file was not created, reason %v", err.Error())
+			}
+
+			helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
```

Comment:
```
Expect(stdOut).ToNot(ContainSubstring("No file changes detected, skipping build"))
```
Add the above assertion, just to make sure push detect the changes and not skipping build.